internal/telemetry/telemetrystore/teestore/teestore.go

```diff
@@ -34,7 +35,7 @@ type Store struct {
 	eventLogs   database.EventLogStore
 }
 
-func NewStore(exportQueue database.TelemetryEventsExportQueueStore, eventLogs database.EventLogStore) *Store {
+func NewStore(exportQueue database.TelemetryEventsExportQueueStore, eventLogs database.EventLogStore) telemetry.EventsStore {
```

Comment:
Most callsites aren't meant to interact with this, I think leaving it as a concrete return is fine:

```suggestion
func NewStore(exportQueue database.TelemetryEventsExportQueueStore, eventLogs database.EventLogStore) *Store {
```

Reply:
You can add a type assertion instead, e.g.

```
var _ telemetry.EventsStore = (*Store)(nil)
```